pkg/cluster/rook_ceph.go

```diff
@@ -355,32 +407,47 @@ func (c *Controller) SetFilesystemReplication(rookVersion semver.Version, name s
 		}
 		return false, errors.Wrapf(err, "get Filesystem %s", name)
 	}
-	changed := false
+	patches := []k8s.JSONPatchOperation{}
 	for i, pool := range fs.Spec.DataPools {
 		current := int(pool.Replicated.Size)
 		if current < level {
 			fs.Spec.DataPools[i].Replicated.Size = uint(level)
-			changed = true
+			patches = append(patches, k8s.JSONPatchOperation{
+				Op:    k8s.JSONPatchOpReplace,
+				Path:  fmt.Sprintf("/spec/dataPools/%d/replicated/size", i),
+				Value: uint(level),
+			})
 		}
 	}
 	current := int(fs.Spec.MetadataPool.Replicated.Size)
 	if current < level {
 		fs.Spec.MetadataPool.Replicated.Size = uint(level)
-		changed = true
+		patches = append(patches, k8s.JSONPatchOperation{
+			Op:    k8s.JSONPatchOpReplace,
+			Path:  "/spec/metadataPool/replicated/size",
+			Value: uint(level),
+		})
 	}
 
-	if !(changed || doFullReconcile) {
+	if !(len(patches) > 0 || doFullReconcile) {
 		return false, nil
```

Comment:
Why the return is false?
Since we are using controller-runtime I'd expected we have the reconcile and be using the return options to restart the Reconcile implemented for it:


- With the error (to do the reconciliation again):`return ctrl.Result{}, err`
- Without an error (to do the reconciliation again to re-check the operations without return an error): `return ctrl.Result{Requeue: true}, nil`
- Therefore, to stop the Reconcile, use:`return ctrl.Result{}, nil`
- Reconcile again after X time: `return ctrl.Result{RequeueAfter: nextRun.Sub(r.Now())}, nil`

For more details, check the Reconcile and its Result[ godoc](https://godoc.org/sigs.k8s.io/controller-runtime/pkg/reconcile).

Reply:
Can you create a story in Replicated's internal Shortcut?

Reply:
Addressed the reasoning for return value in docstring comment